src/app/pages/system/dataset/dataset.component.ts

```diff
@@ -60,30 +60,63 @@ export class DatasetComponent implements OnInit{
     this.ws.call('systemdataset.config').subscribe(res => {
       entityForm.formGroup.controls['pool'].setValue(res.pool);
       entityForm.formGroup.controls['syslog'].setValue(res.syslog);
-      entityForm.formGroup.controls['rrd'].setValue(res.rrd);
     });
   }
 
   customSubmit(value) {
+    this.ws.call("service.query").subscribe(
+      (services) => {
+        const smbShare = _.find(services, {'service': "cifs"});
+        if (smbShare.state === 'RUNNING') {
+          this.dialogService.confirm(
+            T('Restart SMB Service'),
+            T('The SMB service must be restarted after update systemdataset, causing a temporary outage of any active SMB connections. Do you want to continue?'),
```

Comment:
```suggestion
            T('The system dataset has been updated and the SMB service must be restarted. This will cause a temporary disruption of any active SMB connections.'),
```